sycl/plugins/unified_runtime/CMakeLists.txt

```diff
@@ -178,6 +180,13 @@ if ("cuda" IN_LIST SYCL_ENABLE_PLUGINS)
     VERSION "0.0.0"
     SOVERSION "0"
   )
+
+  if(UMF_ENABLE_POOL_TRACKING)
```

Comment:
Actualy we hide it behind the cmake flag because we weren't sure what would be it's performance impact on sycl and we wanted to have a way to quickly disable it in case of a regression. However, we recently run a few benchmarks and it seems like there is no difference in performance. I think we should be fine enabling this flag (maybe we could do this after the release).

Also, we do plan to  optimize the memory tracking pretty soon (right now, it's just a std::map under a lock and we want to use a lock-free structure).

Btw, @kswiecicki do I remember correctly that you are also working on using the umfFree (that relies on tracking in L0)?

Reply:
Yes, I'm experimenting with replacing the existing mechanism for pointer-to-pool matching. Using UMF tracking simplifies the `urUSMFree` code in L0 adapter. There's no need to query for allocation property for deallocation, we no longer need the `SharedReadOnly` etc.